### DISCUSSION TITLE
How to select a `Selectable` with default values

### DISCUSSION THREAD

**Discussion by ISibboI** [diesel-rs/diesel#3739] [Q&A] [answered]
I have a table like this:

```sql
CREATE TABLE job_queue (
    scheduled_time TIMESTAMPTZ NOT NULL,
    name TEXT NOT NULL,
    PRIMARY KEY (scheduled_time, name)
);
```

I would like to create a `Selectable` for this table that automatically populates one field with `CURRENT_TIMESTAMP()`, like this:

```rust
#[derive(Insertable, Queryable, Selectable, Identifiable, AsChangeset, Clone, Debug)]
#[diesel(table_name = crate::schema::job_queue)]
#[diesel(primary_key(scheduled_time, name))]
#[diesel(check_for_backend(diesel::pg::Pg))]
pub struct InProgressJob {
    scheduled_time: DateTime<Utc>,
    start_time: DateTime<Utc>,
    name: String,
}
```

How do I use this as the return type of the following SQL, such that `start_time` will be the result of calling `NOW()`?
```sql
SELECT (scheduled_time, NOW(), name) FROM job_queue LIMIT 10;
```

**Comment by weiznich** [comment, marked as answer]
See the [last example in the documentation](https://docs.diesel.rs/2.1.x/diesel/prelude/trait.Selectable.html#examples). The expression can be freely chosen. The type must match the expression.

**Reply by ISibboI** [reply]
Not sure if I have time for that, but would you accept a pull request explaining this use case in the docs? So far it refers only to "flattening of fields"

**Reply by ISibboI** [reply]
I just opened a pull request #3751
